joins.sql

```diff
@@ -0,0 +1,123 @@
+  FROM posts
+  INNER JOIN users on users.id = posts.user_id
+  where user_id = 200;
+
+--   1. Create a query to get all posts fields, and the user's username, from the `posts` table where the user's first name is 'Norene' and the user's last_name is 'Schmitt'
+
+ SELECT posts.*, users.username
+  FROM posts
+  INNER JOIN users on users.id = posts.user_id
+  where users.first_name = 'Norene' and users.last_name = 'Schmitt';
+
+
+-- 1. Create a query to get usernames from the `users` table where the user has created a post after January 1, 2015
+
+SELECT username
+FROM users
+INNER JOIN posts on users.id = posts.user_id
+WHERE posts.created_at > '2015-01-01';
+
+
+
+-- 1. Create a query to get the post title, post content, and user's username where the user who created the post joined before January 1, 2015
+
+SELECT posts.title, posts.content, users.username
+FROM posts
+INNER JOIN users on users.id = posts.user_id
+WHERE users.created_at < '2015-01-01';
+
+
+
+-- 1. Create a query to get the all rows in the `comments` table, showing post title (aliased as 'Post Title'), and the all the comment's fields
+
+SELECT comments.*, posts.title as "Post Title"
+FROM comments
+INNER JOIN posts on posts.id = comments.post_id;
+
+
+
+
+-- 1. Create a query to get the all rows in the `comments` table, showing post title (aliased as post_title), post url (ailased as post_url), and the comment body (aliased as comment_body) where the post was created before January 1, 2015
+
+SELECT comments.*, posts.title as "post_title", posts.url as "post_url", comments.body as "comment_body"
+FROM comments
+INNER JOIN posts on posts.id = comments.post_id
+WHERE posts.created_at < '2015-01-01';
+
+
+
+-- 1. Create a query to get the all rows in the `comments` table, showing post title (aliased as post_title), post url (ailased as post_url), and the comment body (aliased as comment_body) where the post was created after January 1, 2015
+
+SELECT comments.*, posts.title as "post_title", posts.url as "post_url", comments.body as "comment_body"
+FROM comments
+INNER JOIN posts on posts.id = comments.post_id
+WHERE posts.created_at > '2015-01-01';
+
+
+
+-- 1. Create a query to get the all rows in the `comments` table, showing post title (aliased as post_title), post url (ailased as post_url), and the comment body (aliased as comment_body) where the comment body contains the word 'USB'
+
+SELECT comments.*, posts.title as "post_title", posts.url as "post_url", comments.body as "comment_body"
+FROM comments
+INNER JOIN posts on posts.id = comments.post_id
+WHERE comments.body LIKE '%USB%';
+
+
+
+-- 1. Create a query to get the post title (aliased as post_title), first name of the author of the post, last name of the author of the post, and comment body (aliased to comment_body), where the comment body contains the word 'matrix' ( should have 855 results )
+
+SELECT posts.title as "post_title", users.first_name, users.last_name, comments.body as "comment_body"
```

Comment:
This is actually getting the first and last name of the commenter and not the post author.